### PR TITLE
[Library] Add grouped bar charts

### PR DESCRIPTION
**Pull request by juliawu**
Screenshot 2023-06-28 at 11 54 39 AM
<br />

This PR:
* Adds support for grouped bar charts to the existing bar chart component. 
* Removes the quirk of requiring users to use double quotes within lists by adding a custom attribute converter
* Switches "ES2021" in tsconfig to lowercase so `replaceAll()` can be loaded properly

The above charts were created with:
```
<datacommons-bar
    title="Median income by gender"
    variables='["Median_Income_Person_15OrMoreYears_Male_WithIncome", "Median_Income_Person_15OrMoreYears_Female_WithIncome"]'
    comparisonPlaces='["geoId/01", "geoId/02", "geoId/04"]'
></datacommons-bar>

<datacommons-bar
    title="Median income by gender for counties in Alaska"
    variables="['Median_Income_Person_15OrMoreYears_Male_WithIncome', 'Median_Income_Person_15OrMoreYears_Female_WithIncome']"
    place="geoId/02"
    childPlaceType="County"
></datacommons-bar>
```

Other changes:

### CODE REVIEW

File: static/library/bar_component.ts [hunk 1]
```diff
@@ -72,13 +71,31 @@ export class DatacommonsBarComponent extends LitElement {
...
+  // !Important: variables provided must share the same unit
+  @property({ type: Array<string>, converter: convertArrayAttribute })
+  comparisonVariables;
+
   render(): HTMLElement {
+    const statVarDcids: string[] = this.comparisonVariables
+      ? this.comparisonVariables
+      : [this.variable];
+    const statVarSpec = [];
+    console.log(statVarDcids);
```
Comment (shifucun) [inline static/library/bar_component.ts:88]: remove console.log?

File: static/tsconfig.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
         "name": "typescript-tslint-plugin"
       }
     ],
-    "lib": ["dom", "ES2021"],
+    "lib": ["dom", "es2021"],
```
Comment (shifucun) [inline static/tsconfig.json:8]: nice! can you add a comment for the nuance?

**Review by shifucun — APPROVED** [review]

File: static/library/bar_component.ts (hunk 1 above)
Comment (juliawu) [reply static/library/bar_component.ts]: Thanks for the catch! Removed.

File: static/tsconfig.json (hunk 2 above)
Comment (juliawu) [reply static/tsconfig.json]: Looked into this a bit more, and looks like it was actually a local issue with my VS Code settings (needed to update). Removed this change.

**Pull request merged**